lib/packwerk/association_inspector.rb

```diff
@@ -16,7 +16,10 @@ class AssociationInspector
       has_and_belongs_to_many
     ).to_set
 
-    def initialize(inflector: Inflector.new, custom_associations: Set.new)
+    def initialize(
+      inflector: Inflector.new(custom_inflector: Inflections::Custom.new),
```

Comment:
Why would `association_inspector` know about the specifics of `Inflector`, namely the existence of `Inflections::Custom`?

Reply:
Maybe this should just be the default argument in `Inflector`'s initializer?

Reply:
Mmmh, actually I think there may be a deeper problem here - `Inflections::Custom` doesn't do anything unless fed with an inflections file.

Does it make sense to use empty custom inflections as a default?

Reply:
This was @wildmaples pushing the required argument change to `Packwerk::Inflector#initialize` up to its call sites, i.e. inlining the previous default wherever it was already implicitly used, so there shouldn’t be any change in behaviour here.

I agree that `Inflections::Custom.new` without an inflections file is basically pointless (hence the [suggestion](https://github.com/Shopify/packwerk/pull/53#discussion_r514128192) to remove the default) so I’ll clean this up too. As far as I can tell `AssociationInspector` is only constructed by `RunContext` which [supplies an inflector](https://github.com/Shopify/packwerk/blob/main/lib/packwerk/run_context.rb#L81) anyway so this default is never used in practice.

Reply:
I’ve pushed a revised commit which removes the `inflector` default too. I think “why are we using `Inflections::Custom` without an inflections file” is an interesting design question which we should revisit but I’m choosing not to address it here.